backend/services/scheduler.go

```
@@ -10,82 +14,103 @@
+
+		slog.Debug("Current running jobs for batch", "count", len(jobs), "batchId", batchId)
+
 		if len(jobs) >= maxConcurrencyForBatch {
-			log.Printf("max concurrency for jobs reached: %v, queuing until more jobs succeed", len(jobs))
+			slog.Info("Maximum concurrency reached, queueing job",
+				"jobId", job.DiggerJobID,
+				"currentJobCount", len(jobs),
+				"maxConcurrency", maxConcurrencyForBatch)
+
 			job.Status = orchestrator_scheduler.DiggerJobQueuedForRun
 			models.DB.UpdateDiggerJob(job)
```

golangci-lint (1.64.8)

Error return value of `models.DB.UpdateDiggerJob` is not checked

(errcheck)